Tasks/AndroidSigningV3/androidsigning.ts

```diff
@@ -0,0 +1,135 @@
+import * as path from 'path';
+import * as Q from 'q';
+import * as tl from 'vsts-task-lib/task';
+
+const findAndroidTools = (tool: string): string => {
+    const androidHome = tl.getVariable('ANDROID_HOME');
+    if (!androidHome) {
+        throw tl.loc('AndroidHomeNotSet');
+    }
+
+    // add * in search as on Windows the tool may end with ".exe" or ".bat"
+    const toolsList = tl.findMatch(tl.resolve(androidHome, 'build-tools'), tool + '*', null, { matchBase: true });
+
+    if (!toolsList || toolsList.length === 0) {
+        throw tl.loc('CouldNotFindToolInAndroidHome', tool, androidHome);
+    }
+
+    return toolsList[0];
+};
+
+/*
+Signing the specified file with apksigner.  Move the current file to fn.unsigned, and
+place the signed file at the same location fn
+*/
+const apksigning = (fn: string) => {
+    // file must exist
+    tl.checkPath(fn, 'file to sign');
+
+    let apksigner = tl.getInput('apksignerLocation', false);
+
+    // if the tool path is not set, let's find one (anyone) from the SDK folder
+    if (!apksigner) {
+        apksigner = findAndroidTools('apksigner');
```

Comment:
Do we not want to fall back to JAR signing if the APK signer tool can't be found?

Reply:
We don't want to fall back to JAR signing especially since it has been a year since apksigner switch was made. I like we are updating the major version of the task, so customers who want to use jarsigner can use previous version.

Reply:
I agree.  This is worth considering.  This tool came out with version 24 (Android 7.0).  https://developer.android.com/about/dashboards/ Not many use this, yet.  Developers may not have updated their privately hosted agents either, and want the same definition to work on both.

Reply:
I think it's ok not to fall back if JAR signing is considered deprecated now.  I just wanted to know what the debate was.

Reply:
I think it would be good to add in the "releaseNotes" that if you are on Android 7 or higher, you should use V3. We could update the task description on V2 as well to use V3. I am assuming apksigner would work for older android versions as well if the tool is available and used.